nautobot/extras/tests/test_views.py

```diff
@@ -988,28 +1036,234 @@ def setUpTestData(cls):
     ViewTestCases.ListObjectsViewTestCase,
+    ViewTestCases.DeleteObjectViewTestCase,
+    ViewTestCases.BulkDeleteObjectsViewTestCase,
 ):
     model = JobResult
 
     @classmethod
     def setUpTestData(cls):
         obj_type = ContentType.objects.get(app_label="extras", model="job")
         JobResult.objects.create(
-            name="test1",
+            name="local/test_view/TestJob",
             job_id=uuid.uuid4(),
             obj_type=obj_type,
         )
         JobResult.objects.create(
-            name="test2",
+            name="local/test_view/TestJob2",
             job_id=uuid.uuid4(),
             obj_type=obj_type,
         )
         JobResult.objects.create(
-            name="test3",
+            name="local/test_view/TestJob3",
             job_id=uuid.uuid4(),
             obj_type=obj_type,
         )
+
+
+class JobTestCase(
+    TestCase,
+):
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+    def test_list_without_permission(self):
+        self.assertHttpStatus(self.client.get(reverse("extras:job_list")), 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_list(self):
+        response = self.client.get(reverse("extras:job_list"))
+        self.assertHttpStatus(response, 200)
+
+        response_body = extract_page_body(response.content.decode(response.charset))
+        self.assertIn("DummyJob", response_body)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[], JOBS_ROOT=DUMMY_JOBS)
+    def test_get_without_permission(self):
+        response = self.client.get(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_get(self):
+        response = self.client.get(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 200)
+
+        response_body = extract_page_body(response.content.decode(response.charset))
+        self.assertIn("TestPass", response_body)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[], JOBS_ROOT=DUMMY_JOBS)
+    def test_post_without_permission(self):
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_run_without_schedule(self):
+        self.add_permissions("extras.run_job")
+
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+
+        self.assertHttpStatus(response, 200)
+        errors = extract_form_failures(response.content.decode(response.charset))
+        self.assertEqual(errors, ["_schedule_type: This field is required."])
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_run_now_no_worker(self):
```

Comment:
Will this fail if there *is* a worker running (as is often the case when I'm developing locally)? Do we need to mock.patch the get_worker_count to guarantee that it returns 0 for this test?